integration-test/src/test/java/org/jboss/pnc/integration/endpoints/DeliverableAnalyzerReportEndpointTest.java

```diff
@@ -83,4 +96,84 @@ public void testGetAnalyzedArtifacts() throws ClientException {
+        assertThat(firstLabelHistoryEntry.getLabel()).isEqualTo(DeliverableAnalyzerReportLabel.SCRATCH);
+        assertThat(firstLabelHistoryEntry.getReason()).isEqualTo("This analysis was run as scratch!");
+
+        var secondLabelHistoryEntry = labelHistoryIterator.next();
+        assertThat(secondLabelHistoryEntry.getLabel()).isEqualTo(DeliverableAnalyzerReportLabel.SCRATCH);
+        assertThat(secondLabelHistoryEntry.getReason())
+                .isEqualTo("This was actually quite successful, removing SCRATCH");
+    }
+
+    // TODO: Check also DeliverableAnalyzerReport.labels as part of NCL-8066
+    @Test
+    @InSequence(20)
+    public void testRemoveLabelEntry() throws ClientException {
+        // given
+        var client = new DeliverableAnalyzerReportClient(RestClientConfiguration.asSystem());
+        DeliverableAnalyzerReportLabelRequest request = DeliverableAnalyzerReportLabelRequest.builder()
+                .label(DeliverableAnalyzerReportLabel.SCRATCH)
+                .reason("It's perfect, let's make it RELEASED!")
+                .build();
+
+        // when
+        client.removeLabel(operationId, request);
+        RemoteCollection<DeliverableAnalyzerLabelEntry> labelHistory = client.getLabelHistory(operationId);
+
+        // then
+        assertThat(labelHistory.size()).isEqualTo(3);
+    }
+
+    // TODO: Check also DeliverableAnalyzerReport.labels as part of NCL-8066
+    @Test
+    @InSequence(30)
+    public void testAddLabelEntry() throws ClientException {
+        // given
+        var client = new DeliverableAnalyzerReportClient(RestClientConfiguration.asSystem());
+        DeliverableAnalyzerReportLabelRequest request = DeliverableAnalyzerReportLabelRequest.builder()
+                .label(DeliverableAnalyzerReportLabel.RELEASED)
+                .reason("This was the true diamond, releasing..")
+                .build();
+
+        // when
+        client.addLabel(operationId, request);
+        RemoteCollection<DeliverableAnalyzerLabelEntry> labelHistory = client.getLabelHistory(operationId);
+
+        // then
+        assertThat(labelHistory.size()).isEqualTo(4);
+    }
+
+    @Test
+    @InSequence(40)
+    public void shouldThrowExceptionWhenAddingDuplicate() throws ClientException {
+        // given
+        var client = new DeliverableAnalyzerReportClient(RestClientConfiguration.asSystem());
+        DeliverableAnalyzerReportLabelRequest request = DeliverableAnalyzerReportLabelRequest.builder()
+                .label(DeliverableAnalyzerReportLabel.RELEASED)
+                .reason("Let's try to add duplicate RELEASED ^.-")
+                .build();
+
+        // then
+        try {
+            client.addLabel(operationId, request);
+            fail("Expecting this test method to throw exception when adding duplicate label");
+        } catch (RemoteResourceException ex) {
+            assertThat(ex.getCause().getMessage()).isEqualTo("HTTP 409 Conflict");
+        }
```

Comment:
Please add test that DELETEs a RELEASED report and chceck that the labelHistory got bigger by 2 (remove RELEASED and add DELETED (in that order))